dev/integration_tests/web_e2e_tests/test_driver/cache_width_cache_height_integration.dart

```diff
@@ -0,0 +1,82 @@
+// Copyright 2014 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'dart:async';
+import 'dart:ui' as ui;
+
+import 'package:flutter/material.dart';
+import 'package:flutter_test/flutter_test.dart';
+import 'package:integration_test/integration_test.dart';
+
+
+class LoadTestImageProvider extends ImageProvider<Object> {
```

Comment:
I'm not sure this class is necessary. It's not passed to the `Image` widget as an actual provider. Instead, the test calls its `testLoad` method directly, which means the test could also call `image.image.loadBuffer` directly with the same effect.

Reply:
`image.image.loadBuffer` is a protected method, so `testLoad` was created as a workaround

Reply:
Ah, makes sense. Might be useful to leave a comment explaining this as it's not immediately obvious why this is necessary.

Reply:
will do.